src/backend/distributed/utils/maintenanced.c

```diff
@@ -278,26 +277,14 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 		int latchFlags = WL_LATCH_SET | WL_TIMEOUT | WL_POSTMASTER_DEATH;
 		double timeout = 10000.0; /* use this if the deadlock detection is disabled */
 		bool foundDeadlock = false;
-		time_t currentTime = time(NULL);
-		double secondsSincePrevStatsCollection = difftime(currentTime,
-														  prevStatsCollection);
-		bool citusHasBeenLoaded = false;
 
 		CHECK_FOR_INTERRUPTS();
 
-		StartTransactionCommand();
-		citusHasBeenLoaded = CitusHasBeenLoaded();
-		CommitTransactionCommand();
-
-		if (!citusHasBeenLoaded)
-		{
-			continue;
-		}
-
 		/*
-		 * XXX: We clear the metadata cache before every iteration because otherwise
-		 * it might contain stale OIDs. It appears that in some cases invalidation
-		 * messages for a DROP EXTENSION may arrive during deadlock detection and
+		 * XXX: Each task should clear the metadata cache before every iteration
+		 * by calling InvalidateMetadataSystemCache(), because otherwise it
+		 * might contain stale OIDs. It appears that in some cases invalidation
+		 * messages for a DROP EXTENSION may arrive during these tasks and
 		 * this causes us to cache a stale pg_dist_node OID. We'd actually expect
 		 * all invalidations to arrive after obtaining a lock in LockCitusExtension.
 		 */
```

Comment:
This call to `InvalidateMetadataSystemCache` can be removed

Reply:
Done.